### DISCUSSION TITLE
New project dependency error

### DISCUSSION THREAD

**Discussion by cesar-cardinale** [symfony/symfony#48229] [Q&A] [answered]
### Symfony version(s) affected

6.1.7

### Description

When executing this command :
`symfony new new_project  --webapp`

I get this error :
```
Problem 1
    - symfony/flex[v2.0.0, ..., v2.2.3] require composer-plugin-api ^2.1 -> found composer-plugin-api[2.0.0] but it does not match the constraint.
    - Root composer.json requires symfony/flex ^2 -> satisfiable by symfony/flex[v2.0.0, ..., v2.2.3].
```

What can I do, or need to upgrade to create a new project ?

### How to reproduce

Just run this command `symfony new new_project  --webapp` with composer 2.4.4 you will get the error.

**Comment by nicolas-grekas** [comment]
Hi Cesar, that's because your version of composer is outdated.
Try running `composer self-update`.

**Comment by cesar-cardinale** [comment, marked as answer]
@nicolas-grekas thanks for your answer, but when I do the compose self-update I get :
```
> composer self-update 
You are already using the latest available Composer version 2.4.4 (stable channel).
```

**Reply by nicolas-grekas** [reply]
There is something to figure out on your setup. Composer v2.4 provides a recent enough plugin-api.
Could it be that you have composer/composer in your composer.json file?
Or that you have several versions of composer around?
I converted this to a discussion BTW because I'm not sure there is a bug in Symfony.

**Reply by cesar-cardinale** [reply]
I just re install composer and now it's working, thank you for your help!
